File: C++/Arrays and Hashing/ValidAnagram.cpp
//https://leetcode.com/problems/valid-anagram/

class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size())
        {
            return false;
        }

        unordered_map<char, int> countS;
        unordered_map<char, int> countT;

        for(int i = 0; i < s.size(); i++)
        {
            if(countS.find(s[i]) == countS.end())
            {
                countS[s[i]] = 0;
            }
            if(countT.find(t[i]) == countT.end())
            {
                countT[t[i]] = 0;
            }
            countS[s[i]] += 1;
            countT[t[i]] += 1;
        }
        return countS == countT;
    }
};

File: Python/Arrays and Hashing/ValidAnagram.py
#https://leetcode.com/problems/valid-anagram/

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        letterDictS = {}
        letterDictT = {}
        for letter in s:
            if letter not in letterDictS.keys():
                letterDictS[letter] = 0
            letterDictS[letter] += 1
        for letter in t:
            if letter not in letterDictT.keys():
                letterDictT[letter] = 0
            letterDictT[letter] += 1
        if letterDictS == letterDictT:
            return True
        else:
            return False